<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ProjectExplorer::Internal::DeviceSettingsWidget</class>
 <widget class="QWidget" name="ProjectExplorer::Internal::DeviceSettingsWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>728</width>
    <height>429</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Linux Device Configurations</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QFormLayout" name="formLayout">
       <property name="fieldGrowthPolicy">
        <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
       </property>
       <item row="0" column="0">
        <widget class="QLabel" name="configurationLabel">
         <property name="text">
          <string>&amp;Device:</string>
         </property>
         <property name="buddy">
          <cstring>configurationComboBox</cstring>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="configurationComboBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="editable">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QScrollArea" name="scrollArea">
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>564</width>
          <height>360</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QGroupBox" name="generalGroupBox">
           <property name="title">
            <string>General</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <layout class="QFormLayout" name="formLayout_2">
              <property name="fieldGrowthPolicy">
               <enum>QFormLayout::ExpandingFieldsGrow</enum>
              </property>
              <item row="0" column="0">
               <widget class="QLabel" name="deviceNameLabel">
                <property name="text">
                 <string>&amp;Name:</string>
                </property>
                <property name="buddy">
                 <cstring>nameLineEdit</cstring>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLineEdit" name="nameLineEdit"/>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="oSTypeLabel">
                <property name="text">
                 <string>Type:</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLabel" name="osTypeValueLabel">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="autoDetectionKeyLabel">
                <property name="text">
                 <string>Auto-detected:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QLabel" name="autoDetectionValueLabel">
                <property name="text">
                 <string/>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="deviceStateKeyLabel">
                <property name="text">
                 <string>Current state:</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <layout class="QHBoxLayout" name="horizontalLayout_2">
                <item>
                 <widget class="QLabel" name="deviceStateValueIconLabel"/>
                </item>
                <item>
                 <widget class="QLabel" name="deviceStateValueTextLabel"/>
                </item>
                <item>
                 <spacer name="horizontalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Horizontal</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>40</width>
                    <height>20</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="osSpecificGroupBox">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title">
            <string>Type Specific</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="buttonsLayout">
     <item>
      <widget class="QPushButton" name="addConfigButton">
       <property name="focusPolicy">
        <enum>Qt::StrongFocus</enum>
       </property>
       <property name="text">
        <string>&amp;Add...</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="removeConfigButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="focusPolicy">
        <enum>Qt::StrongFocus</enum>
       </property>
       <property name="text">
        <string>&amp;Remove</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="defaultDeviceButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="focusPolicy">
        <enum>Qt::StrongFocus</enum>
       </property>
       <property name="text">
        <string>Set As Default</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>addConfigButton</sender>
   <signal>clicked()</signal>
   <receiver>ProjectExplorer::Internal::DeviceSettingsWidget</receiver>
   <slot>addDevice()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>818</x>
     <y>35</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>118</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nameLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>ProjectExplorer::Internal::DeviceSettingsWidget</receiver>
   <slot>deviceNameEditingFinished()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>204</x>
     <y>90</y>
    </hint>
    <hint type="destinationlabel">
     <x>330</x>
     <y>332</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeConfigButton</sender>
   <signal>clicked()</signal>
   <receiver>ProjectExplorer::Internal::DeviceSettingsWidget</receiver>
   <slot>removeDevice()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>818</x>
     <y>67</y>
    </hint>
    <hint type="destinationlabel">
     <x>351</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>deviceNameEditingFinished()</slot>
  <slot>deviceTypeChanged()</slot>
  <slot>hostNameEditingFinished()</slot>
  <slot>sshPortEditingFinished()</slot>
  <slot>timeoutEditingFinished()</slot>
  <slot>userNameEditingFinished()</slot>
  <slot>passwordEditingFinished()</slot>
  <slot>addDevice()</slot>
  <slot>selectionChanged()</slot>
  <slot>removeDevice()</slot>
  <slot>authenticationTypeChanged()</slot>
  <slot>keyFileEditingFinished()</slot>
  <slot>testConfig()</slot>
  <slot>deployKey()</slot>
  <slot>gdbServerPortEditingFinished()</slot>
  <slot>currentConfigChanged(int)</slot>
  <slot>showGenerateSshKeyDialog()</slot>
  <slot>showPassword(bool)</slot>
  <slot>handleFreePortsChanged()</slot>
  <slot>showRemoteProcesses()</slot>
 </slots>
</ui>
